File: app/src/main/java/com/example/my_application114514/util/MusicBinder.java
package com.example.my_application114514.util;

import java.util.ArrayList;

import android.os.Binder;

import com.example.my_application114514.data.SongData;
import com.example.my_application114514.listener.PlayerListener;
import com.example.my_application114514.service.MusicService;

public class MusicBinder extends Binder {
  MusicService mMusicService;

  public MusicBinder(MusicService mMusicService) {
    this.mMusicService = mMusicService;
  }

  public void startPlay() {
    mMusicService.startPlay();
  }

  public void updateMusicList(ArrayList<SongData> mList) {
    mMusicService.updateMusicList(mList);
  }

  public void updateMusicCurIndex(int idx) {
    mMusicService.updateMusicCurIndex(idx);
  }

  public boolean isPlaying() {
    return mMusicService.isPlaying();
  }

  public void pause() {
    mMusicService.pause();
  }

  public void play() {
    mMusicService.play();
  }

  public void last() {
    mMusicService.last();
  }

  public void next() {
    mMusicService.next();
  }

  public void stop() {
    mMusicService.stop();
  }

  public SongData getCurSong() {
    return mMusicService.mPlaylist.get(mMusicService.curIndex);
  }

  public int getCurProgress() {
    return mMusicService.getCurProgress();
  }

  public int getEndProgress() {
    return mMusicService.getEndProgress();
  }

  public void seekTo(int progress) {
    mMusicService.seekTo(progress);
  }

  public void setPlayMode(int mode) {
    mMusicService.setPlayMode(mode);
  }

  public void setPlayerListener(PlayerListener playerListener) {
    mMusicService.setPlayerListener(playerListener);
  }
}

File: app/src/main/java/com/example/my_application114514/service/binder/MusicBinder.kt
package com.example.my_application114514.service.binder

import android.os.Binder
import com.example.my_application114514.data.SongData
import com.example.my_application114514.listener.MyPlayerListener
import com.example.my_application114514.service.MusicService

class MusicBinder(private val mMusicService : MusicService) : Binder() {
    fun setMusicList(mSonglist: ArrayList<SongData>?){ mMusicService.setMusicList(mSonglist) }
    fun setMusicCurIndex(curIndex:Int){ mMusicService.setMusicCurIndex(curIndex) }
    fun setPlayerListener(myPlayerListener: MyPlayerListener){ mMusicService.setPlayerListener(myPlayerListener) }
    fun getMusicCurIndex():Int{ return mMusicService.getMusicCurIndex() }
    fun getMusicList():ArrayList<SongData>?{ return mMusicService.getMusicList() }
    fun getCurProgress():Int{ return mMusicService.getCurProgress(); }
    fun getPlayMode():Int{ return mMusicService.getPlayMode() }

    fun startPlay(idx : Int){ mMusicService.startPlay(idx) }
    fun startPlay(){ mMusicService.startPlay() }
    fun isMusicPlaying() : Boolean{ return mMusicService.isMusicPlaying() }
    fun pause(){ mMusicService.pause() }
    fun play(){ mMusicService.play() }
    fun previous(){ mMusicService.previous() }
    fun nextSong(){mMusicService.nextSong()}
    fun stop(){mMusicService.stop()}
    fun seekTo(progress:Int){mMusicService.seekTo(progress)}
    fun switchMode(){mMusicService.switchMode()}
}